### PR TITLE
Feat: First iteration of button improvement

### PR DESCRIPTION
**Pull request by aminpaks**
How to test:
- Head over to styleguide buttons http://localhost:3000/styleguide/buttons

### CODE REVIEW

File: src/Theme/Theme.tsx [hunk 1]
```diff
@@ -5,8 +5,6 @@ import baseStyled, {
 } from 'styled-components';
 import { Theme } from './types';
 
-export { ThemeContext } from 'styled-components';
```
Comment (aminpaks) [inline src/Theme/Theme.tsx]: This didn't work. We needed type assertion that doesn't work in TSX.

**Comment by iliesaithamouda** [comment]
to be honest, i know pretty much nothing in react. so the reviews for me will be more like a learning book and i will sometimes ask questions 😄

**Comment by aminpaks** [comment]
@iliesaithamouda thats alright. We're all learning here.

File: src/Pages/+Shared/Buttons/Util.ts [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+
+export const getBackgroundColor = (state: StyleState = 'normal') => (
+  props: PartialButtonProps & ThemeProps,
+) => {
+  const {
+    variant,
+    theme: { colorSet },
+  } = props;
+  if (state === 'normal') {
+    switch (variant) {
```
Comment (iliesaithamouda) [inline src/Pages/+Shared/Buttons/Util.ts]: why not making an enum type for the variant. the method will be smaller and we can sitch the three if conditions to a switch condition ?..

here's a quick test i did: 

let colorSet: any = {
    primary : '#6783942',
    grayDark : '7686789',
    white : '#fff'

};

enum variant { 
    default = colorSet.primary,
    secondary = colorSet.primary,
    gray = colorSet.grayDark,
    invertOutline = colorSet.white,
    castOutline = colorSet.secondary
}

let test: number = 1;
let testVariant: string = 'default'

switch (test) {
    case 1: console.log(variant[testVariant]);
}

Comment (iliesaithamouda) [reply src/Pages/+Shared/Buttons/Util.ts]: the exemple is to test if what i'm sayin makes sens....lol

Comment (aminpaks) [reply src/Pages/+Shared/Buttons/Util.ts]: The conditionals for state not the variants. There's 3 state of `normal`, `hover` and `focus`.

Comment (aminpaks) [reply src/Pages/+Shared/Buttons/Util.ts]: Also the enum that you suggested is a mapping mechanism, but the color values are not available till runtime. I don't see how we can improve the mapping as these utility functions are just mapping functions.

Comment (iliesaithamouda) [reply src/Pages/+Shared/Buttons/Util.ts]: oh true for the color values available on runtime 👍 
for the conditional of state, i meant that the state can be in a switch condition and simply return the value of the variant enum (if we had the color before the runtime of course...) 
exemple: 
switch(state) {
     case 'hover': return variant[variant]
}

or am i missing something ?

Comment (iliesaithamouda) [reply src/Pages/+Shared/Buttons/Util.ts]: nevermind.... i just understood what you're doing...i must be tired... forget my comments lolll

Comment (aminpaks) [reply src/Pages/+Shared/Buttons/Util.ts]: Yeah we could break these function into 3 and call them from one. That would actually improve visibility a bit.

File: src/Pages/+Shared/Buttons/Util.ts [hunk 3]
```diff
@@ -0,0 +1,236 @@
+import { PartialButtonProps } from './Button';
+import { ThemeProps, whiten, darken, fade, lighten } from '../../../Theme';
+type StyleState = 'normal' | 'hover' | 'focus';
+
+export const getBackgroundColor = (state: StyleState = 'normal') => (
+  props: PartialButtonProps & ThemeProps,
```
Comment (nasreddineskandrani) [inline src/Pages/+Shared/Buttons/Util.ts]: you can make `PartialButtonProps` extends `ThemeProps` and you ll not have to do it each time

Comment (aminpaks) [reply src/Pages/+Shared/Buttons/Util.ts]: done.

**Pull request merged**